File: src/main/java/no/nav/pensjon/regler/domain/beregning/penobjekter/MotregningYtelseskomponent.java
package no.nav.pensjon.regler.domain.beregning.penobjekter;

/**
 * Created by N123422 on 06.11.2014.
 */
public abstract class MotregningYtelseskomponent extends BeregningYtelseskomponent {

    private static final long serialVersionUID = -675866734547173660L;
    private Integer dagsats = 0;
    private Integer antallDager = 0;

    public MotregningYtelseskomponent() {
        super();
    }

    public MotregningYtelseskomponent(MotregningYtelseskomponent motregningYtelseskomponent) {
        super(motregningYtelseskomponent);
        dagsats = motregningYtelseskomponent.dagsats;
        antallDager = motregningYtelseskomponent.antallDager;
    }

    public Integer getDagsats() {
        return dagsats;
    }

    public void setDagsats(Integer dagsats) {
        this.dagsats = dagsats;
    }

    public Integer getAntallDager() {
        return antallDager;
    }

    public void setAntallDager(Integer antallDager) {
        this.antallDager = antallDager;
    }
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning/penobjekter/MotregningYtelseskomponent.kt
package no.nav.pensjon.regler.domain.beregning.penobjekter

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo

/**
 * Created by N123422 on 06.11.2014.
 */
@JsonSubTypes(
    JsonSubTypes.Type(value = ArbeidsavklaringspengerUT::class),
    JsonSubTypes.Type(value = Arbeidsavklaringspenger::class),
    JsonSubTypes.Type(value = Sykepenger::class),
    JsonSubTypes.Type(value = SykepengerUT::class),
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
abstract class MotregningYtelseskomponent : BeregningYtelseskomponent() {
    var dagsats = 0
    var antallDager = 0
}